xbmc/threads/Thread.cpp

```diff
@@ -20,6 +20,7 @@
  *
  */
 
+#include "Application.h"
```

Comment:
this needs a different solution. a lower layer must not have such kind of dependency to upper ones.